transport/node.go

```diff
@@ -0,0 +1,29 @@
+package transport
+
+import (
+	"go.uber.org/atomic"
+)
+
+type nodeStatus = *atomic.Bool
```

Comment:
The pointer here is not needed, unless all APIs use Node by copy.
From what I see we have *Node, in that case compiler takes pointer from the struct on heap, no need for a dedicated pointer here.